tests/helpers.py

```diff
@@ -1071,3 +1084,206 @@ def verify_pv_mounted_on_node(node_pv_dict):
+        size (str): size of the pvc eg: '10Gi'
+        access_modes (list): List of access modes for PVC creation
+
+    Returns:
+        pvc_objs_list (list): List of pvc objs created in function
+    """
+    obj_status_list, result_lists = ([] for i in range(2))
+    with ThreadPoolExecutor() as executor:
+        for mode in access_modes:
+            result_lists.append(
+                executor.submit(
+                    create_multiple_pvcs, sc_name=sc_obj.name,
+                    namespace=namespace, number_of_pvc=number_of_pvc,
+                    access_mode=mode, size=size)
+            )
+    result_list = [result.result() for result in result_lists]
+    pvc_objs_list = converge_lists(result_list)
+    # Check for all the pvcs in Bound state
+    with ThreadPoolExecutor() as executor:
+        for objs in pvc_objs_list:
+            obj_status_list.append(
+                executor.submit(wait_for_resource_state, objs, 'Bound')
+            )
+    if False in [obj.result() for obj in obj_status_list]:
+        raise TimeoutExpiredError
+    return pvc_objs_list
+
+
+def create_pods_parallel(pvc_list, namespace, interface):
+    """
+    Function to create pods in parallel
+
+    Args:
+        pvc_list (list): List of pvcs to be attached in pods
+        namesapce (str): The namespace for creating pod
+        interface (str): The interface backed the PVC
+
+    Returns:
+        pod_objs (list): Returns list of pods created
+    """
+    future_pod_objs = []
+    # Added 300 sec wait time since in scale test once the setup has more
+    # PODs time taken for the pod to be up will be based on resource available
+    wait_time = 300
+    with ThreadPoolExecutor() as executor:
+        for pvc_obj in pvc_list:
+            future_pod_objs.append(executor.submit(
+                create_pod, interface_type=interface,
+                pvc_name=pvc_obj.name, do_reload=False, namespace=namespace)
+            )
+    pod_objs = [pvc_obj.result() for pvc_obj in future_pod_objs]
+    # Check for all the pods are in Running state
+    # In above pod creation not waiting for the pod to be created because of threads usage
+    with ThreadPoolExecutor() as executor:
+        for obj in pod_objs:
+            future_pod_objs.append(
+                executor.submit(wait_for_resource_state, obj, 'Running', timeout=wait_time)
+            )
+    # If pods not up raise exception/failure
+    if False in [obj.result() for obj in future_pod_objs]:
+        raise TimeoutExpiredError
+    return pod_objs
+
+
+def delete_objs_parallel(obj_list):
+    """
+    Function to delete objs specified in list
+    Args:
+        obj_list(list): List can be obj of pod, pvc, etc
+
+    Returns:
+        bool: True if obj deleted else False
+
+    """
+    with ThreadPoolExecutor() as p:
+        for obj in obj_list:
+            p.submit(obj.delete)
+    return True
+
+
+def memory_leak_analysis(median_dict):
```

Comment:
I think this function should be under ocs_ci.ocs.resources.node

Reply:
This function has no dependency on nodes, it requires the top output file in /tmp/ location for each node. which will be created by memory_leak_function(). In code am getting to worker nodes to use the names. I would suggest to have this function part of the helpers.